walkthroughs/week-5-tdd/intro/src/main/java/com/google/sps/Greeter.java

```diff
@@ -22,6 +22,12 @@
    * Returns a greeting for the given name.
    */
   public String greet(String name) {
-    return "Hello " + name;
+    String cleanedName = "";
+    char [] nameRay = name.trim().toCharArray();
+    for(char c : nameRay){
+        cleanedName += (Character.isLetter(c) || Character.isDigit(c))? c : ""; 
```

Comment:
I believe there is a Character.isAlphaNumeric?

Also, make sure to space the operators. 

Finally, Use a StringBuilder for cleanedName

if (...) {
  cleanedName.append(c);
}